storage/aws/aws_test.go

```diff
@@ -0,0 +1,410 @@
+// Copyright 2024 The Tessera authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This the tests for a MySQL+S3 AWS Tessera implementation.  It requires a
+// MySQL database to successfully run the MySQL tests, otherwise they are
+// skipped.  Run tests with `-parallel=1` to avoid concurent tests on the same
+// database, and specifically runs of `mustDropTables`.
+//
+// Sample command to start a local MySQL database using Docker:
+// $ docker run --name test-mysql -p 3306:3306 -e MYSQL_ROOT_PASSWORD=root -e MYSQL_DATABASE=test_tessera -d mysql
+package aws
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"database/sql"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"reflect"
+	"sync"
+	"testing"
+
+	gcs "cloud.google.com/go/storage"
+	"github.com/aws/smithy-go"
+	"github.com/google/go-cmp/cmp"
+	tessera "github.com/transparency-dev/trillian-tessera"
+	"github.com/transparency-dev/trillian-tessera/api"
+	"github.com/transparency-dev/trillian-tessera/api/layout"
+	storage "github.com/transparency-dev/trillian-tessera/storage/internal"
+	"k8s.io/klog"
+)
+
+var (
+	mySQLURI            = flag.String("mysql_uri", "root:root@tcp(localhost:3306)/test_tessera", "Connection string for a MySQL database")
+	isMySQLTestOptional = flag.Bool("is_mysql_test_optional", true, "Boolean value to control whether the MySQL test is optional")
+)
+
+// TestMain parses flags.
+func TestMain(m *testing.M) {
+	klog.InitFlags(nil)
+	flag.Parse()
```

Comment:
You _may_ not need the call to `flag.Parse` here as `M.Run()` will parse flags if they've not already been parsed: https://cs.opensource.google/go/go/+/refs/tags/go1.23.3:src/testing/testing.go;l=1942-1945